blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -57,47 +50,71 @@ public ElasticIO(ElasticClientManager manager) {
 
 
     public void insertDiscovery(IMetric metric) throws IOException {
-        List<IMetric> batch = new ArrayList<IMetric>();
+        List<IMetric> batch = new ArrayList<>();
         batch.add(metric);
         insertDiscovery(batch);
     }
 
-    public void insertDiscovery(List<IMetric> batch) throws IOException {
+    // REST call to index into ES
+    public void insertDiscovery(List<IMetric> batch) {
         batchHistogram.update(batch.size());
         if (batch.size() == 0) {
             return;
         }
-        
-        // TODO: check bulk insert result and retry
+
         Timer.Context ctx = writeTimer.time();
         try {
-            BulkRequestBuilder bulk = client.prepareBulk();
             for (Object obj : batch) {
                 if (!(obj instanceof IMetric)) {
                     classCastExceptionMeter.mark();
                     continue;
                 }
-
-                IMetric metric = (IMetric)obj;
-                Locator locator = metric.getLocator();
-                Discovery discovery = new Discovery(locator.getTenantId(), locator.getMetricName());
-
-                Map<String, Object> fields = new HashMap<String, Object>();
-
-
-                if (obj instanceof  Metric && getUnit((Metric)metric) != null) { // metric units may be null
-                    fields.put(ESFieldLabel.unit.toString(), getUnit((Metric) metric));
-                }
-
-                discovery.withSourceFields(fields);
-                bulk.add(createSingleRequest(discovery));
             }
-            bulk.execute().actionGet();
+            elasticsearchRestHelper.indexMetrics(batch);
+        } catch (IOException e) {
+            log.error(e.getMessage());
```

Comment:
Since, there was no logging at all earlier in any of these classes, I added that to make sure at least we are logging something. :)
I will add more to it in all of the classes metrics and event based.

Reply:
done